integration/pvtdata/pvtdata_test.go

```diff
@@ -230,38 +358,90 @@ var _ bool = Describe("PrivateData", func() {
 				Policy:            `OR ('Org1MSP.member','Org2MSP.member', 'Org3MSP.member')`,
 				CollectionsConfig: filepath.Join("testdata", "collection_configs", "collections_config1.json")}
 
-			nwo.InstallChaincode(network, chaincode, org2peer1)
+			sess, err = network.PeerUserSession(org2peer1, "Admin2", commands.ChaincodeInstall{
+				Name:        chaincode.Name,
+				Version:     chaincode.Version,
+				Path:        chaincode.Path,
+				Lang:        chaincode.Lang,
+				PackageFile: chaincode.PackageFile,
+			})
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess, network.EventuallyTimeout).Should(gexec.Exit(0))
+
+			sess, err = network.PeerUserSession(org2peer1, "Admin2", commands.ChaincodeListInstalled{})
+			Expect(err).NotTo(HaveOccurred())
+			Eventually(sess, network.EventuallyTimeout).Should(gexec.Exit(0))
+			Expect(sess).To(gbytes.Say(fmt.Sprintf("Name: %s, Version: %s,", chaincode.Name, chaincode.Version)))
 
 			expectedPeers = []*nwo.Peer{
 				network.Peer("org1", "peer0"),
 				network.Peer("org2", "peer0"),
 				network.Peer("org2", "peer1"),
-				network.Peer("org3", "peer0")}
+				network.Peer("org3", "peer0"),
+			}
 
-			By("verifying membership")
-			verifyMembership(network, expectedPeers, "testchannel", "marblesp")
+			By("making sure all peers have the same ledger height")
+			for _, peer := range expectedPeers {
+				Eventually(func() int {
+					var (
+						sess *gexec.Session
+						err  error
+					)
+					if peer.ID() == "org2.peer1" {
+						// use Admin2 user for peer1.org2
+						sess, err = network.PeerUserSession(peer, "Admin2", commands.ChannelInfo{
+							ChannelID: "testchannel",
+						})
+						Expect(err).NotTo(HaveOccurred())
+						Eventually(sess, network.EventuallyTimeout).Should(gexec.Exit(0))
+					} else {
+						sess, err = network.PeerUserSession(peer, "User1", commands.ChannelInfo{
```

Comment:
if you have "getLedgerHeight" why do you need this "ChannelInfo" command? can't you use the getLedgerheight ?

Reply:
getLedgerHeight specifically uses `User1` for the peer user session. Unfortunately any operations on the new peer must use the `Admin2` user so a lot of the logic in this integration test revolves around passing in a custom user for the peer user session commands that need to be run on `org2.peer1`. Because this seemed like a pretty isolated test case I figured it wasn't worth extending the functions to take a user in and just leave them to default to `User1`

Reply:
1. Why can't you use User1 ? 
2. Can't you add a parameter to the **getLedgerHeight** function to distinguish?

Reply:
User1 certs can't be replaced because org2.peer0 still uses them, instead I copied the generated admin user over as Admin2 specifically to use for org2.peer1 operations. I don't mind extending `getLedgerHeight` but figured it wasn't worth it for just this use case.